corehq/apps/app_manager/static/app_manager/js/app_view.js

```diff
@@ -29,17 +29,20 @@ hqDefine("app_manager/js/app_view", function () {
             // prepend with blank so placeholder works
             self.domainNames = [''].concat(data("domain_names"));
             self.linkableDomains = data("linkable_domains");
+            self.shouldLimitToLinkedDomains = data("limit_to_linked_domains");
 
             self.isChecked = ko.observable(false);
             self.shouldEnableLinkedAppOption = ko.observable(true);
 
             self.domainChanged = function (data, event) {
-                var selectedDomain = event.target.options[event.target.selectedIndex].value;
-                self.shouldEnableLinkedAppOption(self.linkableDomains.includes(selectedDomain));
+                if (self.shouldLimitToLinkedDomains) {
+                    var selectedDomain = event.target.options[event.target.selectedIndex].value;
```

Comment:
Didn't notice on previous review: using `currentTarget` is a good habit, since `target` can be a descendant of the element that's listening for the event: https://developer.mozilla.org/en-US/docs/Web/API/Event/currentTarget